File: apps/messenger/src/modules/discord/entities/discord-message-author.entity.ts
import { MessengerBaseEntity } from "src/database/base/messenger-base.entity";
import { DiscordMessageEntity } from "./discord-message.entity";

export interface DiscordMessageAuthorEntity extends MessengerBaseEntity<DiscordMessageAuthorEntity> {
  id: number;
  avatarPngUrl?: string;
  bannerPngUrl?: string;
  bot: boolean;
  authorCreatedAt: Date;
  discriminator: string;
  displayName: string;
  globalName?: string;
  authorId: string;
  tag: string;
  system: boolean;
  username: string;
  message: DiscordMessageEntity;
}

File: apps/builder/src/main/kotlin/dev/felipeflohr/w2gservices/builder/entities/DiscordMessageAuthorEntity.kt
package dev.felipeflohr.w2gservices.builder.entities

import dev.felipeflohr.w2gservices.builder.annotations.NoArg
import dev.felipeflohr.w2gservices.builder.base.BuilderBaseEntity
import java.time.Instant
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table

@NoArg
@Table("\"TB_DISCORD_MESSAGE_AUTHOR\"")
class DiscordMessageAuthorEntity(
    @Id
    @Column("\"DMA_ID\"")
    var id: Long? = null,

    @Column("\"DMA_AVAPNGURL\"")
    var avatarPngUrl: String?,

    @Column("\"DMA_BANPNGURL\"")
    var bannerPngUrl: String?,

    @Column("\"DMA_BOT\"")
    var bot: Boolean? = false,

    @Column("\"DMA_AUT_CREATED_AT\"")
    var autCreatedAt: Instant,

    @Column("\"DMA_DIS_NAME\"")
    var displayName: String,

    @Column("\"DMA_GNAME\"")
    var globalName: String?,

    @Column("\"DMA_AUTID\"")
    var authorId: String,

    @Column("\"DMA_SYS\"")
    var system: Boolean? = false,

    @Column("\"DMA_USERNAME\"")
    var username: String,
) : BuilderBaseEntity()
